File: packages/sdui/src/main/java/nl/q42/sdui/screen/ProfileScreen.java
package nl.q42.sdui.screen;

import org.springframework.stereotype.Component;

import nl.q42.common.screen.Screen;

@Component
public class ProfileScreen extends Screen
{

  public static final String SCREEN_NAME = "profile";

  public ProfileScreen()
  {
    super(SCREEN_NAME);
  }
}

File: packages/sdui/src/main/kotlin/nl/q42/sdui/screen/ProfileScreen.kt
package nl.q42.sdui.screen

import kotlinx.serialization.Serializable
import nl.q42.common.ServerComponent
import nl.q42.common.TextComponent
import nl.q42.core.AppRequestContext

@org.springframework.stereotype.Component
@Serializable
class ProfileScreen : ScreenInstance {

    override fun content(context: AppRequestContext): List<ServerComponent> {
        return listOf(
            TextComponent("test", contentId = "text-comp")
        )
    }

    override fun name(): String {
        return SCREEN_IDENTIFIER
    }

    companion object {
        const val SCREEN_IDENTIFIER: String = "profile"
    }
}
